qcodes/instrument_drivers/tektronix/AWG5014.py

```diff
@@ -787,367 +1055,324 @@ def generate_awg_file(self,
+        """
         self.write('WLISt:WAVeform:DELete ALL')
 
-    # Ask for string with filenames
     def get_filenames(self):
-        # logging.debug(__name__ + ' : Read filenames from instrument')
-        return self.ask('MMEM:CAT?')
-
-    def set_DC_out(self, DC_channel_number, Voltage):
-        self.write('AWGControl:DC%s:VOLTage:OFFSet %sV' %
-                   (DC_channel_number, Voltage))
-
-    def get_DC_out(self, DC_channel_number):
-        return self.ask('AWGControl:DC%s:VOLTage:OFFSet?' % DC_channel_number)
-
-    def send_DC_pulse(self, DC_channel_number, Amplitude, length):
-        '''
-        sends a (slow) pulse on the DC channel specified
-        Ampliude: voltage level
-        length: seconds
-        '''
-        restore = self.get_DC_out(DC_channel_number)
-        self.set_DC_out(DC_channel_number, Amplitude)
+        """Duplicate of self.get_folder_contents"""
+        return self.ask('MMEMory:CATalog?')
+
+    def send_DC_pulse(self, DC_channel_number, set_level, length):
+        """
+        Sets the DC level on the specified channel, waits a while and then
+        resets it to what it was before.
+
+        Note: Make sure that the output DC state is ON.
+
+        Args:
+            DC_channel_number (int): The channel number (1-4).
+            set_level (float): The voltage level to set to (V).
+            length (flaot): The time to wait before resetting (s).
+        """
+        DC_channel_number -= 1
+        chandcs = [self.ch1_DC_out, self.ch2_DC_out, self.ch3_DC_out,
+                   self.ch4_DC_out]
+
+        restore = self.chandcs[DC_channel_number].get()
+        self.chandcs[DC_channel_number].set(set_level)
         sleep(length)
-        self.set_DC_out(DC_channel_number, restore)
-
-    def set_DC_state(self, state=False):
-        self.write('AWGControl:DC:state %s' % (int(state)))
-
-    def get_DC_state(self):
-        return self.ask('AWGControl:DC:state?')
-
-    # Send waveform to the device (from transmon driver)
-
-    def upload_awg_file(self, fname, fcontents):
-        t0 = time()
-        self._rem_file_path
-        floc = self._rem_file_path
-        f = open(floc + '\\' + fname, 'wb')
-        f.write(fcontents)
-        f.close()
-        t1 = time() - t0
-        print('upload time: ', t1)
-        self.get_state()
-        print('setting time: ', time() - t1 - t0)
-
-    def _set_setup_filename(self, fname):
-        folder_name = 'C:/' + self._setup_folder + '/' + fname
-        self.set_current_folder_name(folder_name)
-        set_folder_name = self.get_current_folder_name()
-        if not os.path.split(folder_name)[1] == os.path.split(set_folder_name)[1][:-1]:
-            print('Warning, unsuccesfully set AWG file', folder_name)
-        print('Current AWG file set to: ', self.get_current_folder_name())
-        self.write('AWGC:SRES "%s.awg"' % fname)
-
-    def set_setup_filename(self, fname, force_load=False):
```

Comment:
My idea was that it is better to have just one way of doing things. To load an awg file from memory (which is not a stupid thing to do for large files where file transfer times matter), one can/should now use `load_awg_file`. Do you think that `set_setup_filename` does something very different than `load_awg_file`? And if so, I would suggest to incorporate that into `load_awg_file`.